src/components/forms/ProductSelect/ProductSelect.js

```diff
@@ -0,0 +1,51 @@
+import PropTypes from 'prop-types';
+
+import AsyncSelectField from '../SelectField/AsyncSelectField';
+
+const ProductSelect = ({namePrefix, label, description, isRequired}) => {
+  const getOptions = async () => {
+    // Mocking the response
+    const response = await new Promise(resolve => {
+      setTimeout(() => {
+        resolve([
+          {
+            code: 'P1',
+            identifier: 'product1',
+            name: 'Product 1',
+          },
+          {
+            code: 'P2',
+            identifier: 'product2',
+            name: 'Product 2',
+          },
+          {
+            code: 'P3',
+            identifier: 'product3',
+            name: 'Product 3',
+          },
+        ]);
+      }, 1000);
+    });
+    return response.map(({code, identifier, name}) => ({
+      label: `${code} - ${name}`,
+      value: identifier,
+    }));
```

Comment:
We don't need to remap this, you can instead specify `valueProperty="identifier"` on the `<AsyncSelect>` component.

For the label, I think only `name` is relevant, looking at the existing dropdown:

![image](https://github.com/open-formulieren/open-forms-sdk/assets/5518550/0d65d4e4-d732-4a65-8f79-0b9ce5e340cb)